src/components/registerpage/Register/index.jsx

```diff
@@ -0,0 +1,253 @@
+    password: "",
+    isAgreed: false,
+  });
+  const [errors, setErrors] = useState({});
+  const [isTermsOpen, setIsTermsOpen] = useState(false);
+  const handleChange = (key, value) => {
+    const newValues = { ...registerInformation };
+    newValues[key] = value;
+    setRegisterInformation(newValues);
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const { fullName, email, age, password, isAgreed } = registerInformation;
+    const newErrors = {};
+    if (fullName.trim() === "") {
+      newErrors.fullName = t("register.fullName");
+    }
+    if (age.trim() === "") {
+      newErrors.age = t("register.age");
+    }
+    if (password.trim().length < MAX_PASSWORD_LENGTH) {
+      newErrors.password = t("register.passwordCharacter");
+    }
+    if (password.trim() === "") {
+      newErrors.password = t("register.enterPassword");
+    }
+    if (validateEmail(email).length === 0) {
+      newErrors.email = t("register.emailFormat");
+    }
+    if (!isAgreed) {
+      newErrors.isAgreed = t("register.agreeOnTerms");
+    }
+
+    if (Object.keys(newErrors).length > 0) {
+      setErrors(newErrors);
+      return;
+    }
+    onSubmit(registerInformation);
+  };
+
+  const handleOkTerms = () => {
+    handleChange("isAgreed", true);
+    setIsTermsOpen(false);
+  };
+  const handleCancelTerms = () => {
+    setIsTermsOpen(false);
+  };
+  const termsAndConditionsTranslation = [
+    ...t("register.terms", { returnObjects: true }),
+  ];
+
+  return (
+    <form noValidate onSubmit={handleSubmit}>
+      <Modal
+        title={t("register.termsTitle")}
+        visible={isTermsOpen}
+        onOk={handleOkTerms}
+        onCancel={handleCancelTerms}
+        footer={[
+          <Button key="back" onClick={handleCancelTerms}>
+            {t("register.close")}
+          </Button>,
+          <Button key="submit" type="primary" onClick={handleOkTerms}>
+            {t("register.iAgree")}
+          </Button>,
+        ]}
+      >
+        {t("register.agreeOnTerms", { returnObjects: true })}
+      </Modal>
+      <div className="loginContainer">
+        <div className="loginContainer__loginTitle">{t("register.create")}</div>
+        {error && (
+          <Alert
+            style={{ marginBottom: 10 }}
+            type="error"
+            showIcon
+            message={error}
+          />
+        )}
+        <div className="loginContainer__loginDialog">
+          <div className="loginContainer__loginDialog__inputLabel">
+            {t("register.name")}
+          </div>
+          <div className="loginContainer__loginDialog__input">
+            <Input
+              className={
+                errors.fullname &&
+                "loginContainer__loginDialog__input__hasError"
+              }
+              value={registerInformation.fullname}
+              onChange={(e) => {
+                handleChange("fullname", e.target.value);
+              }}
+            />
+            {errors.fullname && (
```

Comment:
same, lines 121 to 125 are basically copy-pasted everywhere

Reply:
PTAL